core/src/types/address.rs

```diff
@@ -254,6 +259,9 @@ impl Address {
                 debug_assert_eq!(string.len(), FIXED_LEN_STRING_BYTES);
                 string
             }
+            Address::Foreign(addr) => {
+                format!("{}::{}", PREFIX_FOREIGN, addr)
+            }
```

Comment:
I think this is buggy, as we require these strings to be of a fixed length (45 bytes iirc?). bech32 decoding will fail for foreign addresses, unless we change the decoding logic